src/core/frontend/emu_window.cpp

```diff
@@ -21,21 +21,19 @@ class EmuWindow::TouchState : public Input::Factory<Input::TouchDevice>,
 
     std::mutex mutex;
 
-    bool touch_pressed = false; ///< True if touchpad area is currently pressed, otherwise false
-
-    float touch_x = 0.0f; ///< Touchpad X-position
-    float touch_y = 0.0f; ///< Touchpad Y-position
+    Input::TouchStatus status;
 
 private:
     class Device : public Input::TouchDevice {
     public:
         explicit Device(std::weak_ptr<TouchState>&& touch_state) : touch_state(touch_state) {}
-        std::tuple<float, float, bool> GetStatus() const override {
+        Input::TouchStatus GetStatus() const override {
+            Input::TouchStatus touch_status{};
             if (auto state = touch_state.lock()) {
                 std::lock_guard guard{state->mutex};
-                return std::make_tuple(state->touch_x, state->touch_y, state->touch_pressed);
+                touch_status = state->status;
```

Comment:
```suggestion
                return state->status;
```

This results in a redundant copy before the return otherwise.